File: src/java/sample/xt/sample/PrintSimple.java
package xt.sample;

import java.util.EnumSet;
import xt.audio.Enums.XtEnumFlags;
import xt.audio.Enums.XtSystem;
import xt.audio.XtAudio;
import xt.audio.XtDeviceList;
import xt.audio.XtPlatform;
import xt.audio.XtService;

public class PrintSimple {

    public static void main() throws Exception {
        try(XtPlatform platform = XtAudio.init(null, null)) {
            for(XtSystem system: platform.getSystems()) {
                XtService service = platform.getService(system);
                try(XtDeviceList list = service.openDeviceList(EnumSet.of(XtEnumFlags.ALL))) {
                    for(int d = 0; d < list.getCount(); d++) {
                        String id = list.getId(d);
                        System.out.println(system + ": " + list.getName(id));
                    }
                }
            }
        }
    }
}

File: src/cli/sample/PrintSimple.cs
using System;

namespace Xt
{
    public static class PrintSimple
    {
        public static void Main()
        {
            using XtPlatform platform = XtAudio.Init(null, IntPtr.Zero, null);
            foreach (XtSystem system in platform.GetSystems())
            {
                XtService service = platform.GetService(system);
                XtDeviceList list = service.OpenDeviceList(XtEnumFlags.All);
                for (int d = 0; d < list.GetCount(); d++)
                {
                    string id = list.GetId(d);
                    Console.WriteLine(system + ": " + list.GetName(id));
                }
            }
        }
    }
}